drivers/adc/adc_mcux_lpadc.c

```diff
@@ -200,8 +206,25 @@ static int mcux_lpadc_channel_setup(const struct device *dev,
 	}
 #endif
 
-	if (channel_cfg->reference != ADC_REF_EXTERNAL0) {
-		LOG_ERR("Invalid channel reference");
+	/*
+	 * ADC_REF_EXTERNAL1: Use SoC internal regulator as LPADC reference voltage.
+	 * ADC_REF_EXTERNAL0: Use other voltage source (maybe also within the SoCs)
+	 * as LPADC reference voltage, like VREFH, VDDA, etc.
+	 */
+	if (channel_cfg->reference == ADC_REF_EXTERNAL1) {
```

Comment:
do we need to add more options to the ADC ref enum in the api header?

Reply:
Maybe not, and I would like to change the **ADC_REF_EXTERNAL1** to **ADC_REF_INTERNAL** and add **ref_internal** for the LPADC driver.